Ginger/GingerCoreNET/Run/Excels/ExcelNPOIOperations.cs

```diff
@@ -0,0 +1,341 @@
+        DataTable FilteredDataTable { get; set; }
+        List<Tuple<string, object>> UpdateCellList { get; set; }
+        IWorkbook Workbook { get; set; }
+        ISheet Sheet { get; set; }
+
+        private DataTable ConvertSheetToDataTable(ISheet sheet)
+        {
+            try
+            {
+                var dtExcelTable = new DataTable();
+                dtExcelTable.Rows.Clear();
+                dtExcelTable.Columns.Clear();
+                var headerRow = sheet.GetRow(0);
+                int colCount = headerRow.LastCellNum;
+                for (var c = 0; c < colCount; c++)
+                {
+                    if (!dtExcelTable.Columns.Contains(headerRow.GetCell(c).ToString()))
+                    {
+                        dtExcelTable.Columns.Add(headerRow.GetCell(c).ToString());
+                    }
+                }
+                var i = 1;
+                var currentRow = sheet.GetRow(i);
+                while (currentRow != null)
+                {
+                    var dr = dtExcelTable.NewRow();
+                    for (var j = 0; j < dr.ItemArray.Length; j++)
+                    {
+                        var cell = currentRow.GetCell(j);
+
+                        if (cell != null)
+                        {
+                            switch (cell.CellType)
+                            {
+                                case CellType.Numeric:
+                                    dr[j] = DateUtil.IsCellDateFormatted(cell)
+                                        ? cell.DateCellValue.ToString(CultureInfo.InvariantCulture)
+                                        : cell.NumericCellValue.ToString(CultureInfo.InvariantCulture);
+                                    break;
+                                case CellType.String:
+                                    dr[j] = cell.StringCellValue;
+                                    break;
+                                case CellType.Boolean:
+                                    dr[j] = cell.BooleanCellValue;
+                                    break;
+                                case CellType.Formula:
+                                    dr[j] = cell.CellFormula;
+                                    break;
+                                case CellType.Blank:
+                                    dr[j] = string.Empty;
+                                    break;
+                                default:
+                                    dr[j] = cell.RichStringCellValue;
+                                    break;
+                            }
+                        }
+                    }
+                    dtExcelTable.Rows.Add(dr);
+                    i++;
+                    currentRow = sheet.GetRow(i);
+                }
+                return dtExcelTable;
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Can't convert sheet to data table, " + ex.Message);
```

Comment:
No need to catch and throw, just let the exception get to the higher caller to be handeled

Reply:
done